packages/augur-sdk/src/api/Gnosis.ts

```diff
@@ -1,77 +1,401 @@
+import { Address } from '../state/logs/types';
+
+export const AUGUR_GNOSIS_SAFE_NONCE = 872838000000;
+
+export interface CreateGnosisSafeViaRelayParams {
+  owner: string;
+  paymentToken: string;
+}
+
+export interface CalculateGnosisSafeAddressParams
+  extends CreateGnosisSafeViaRelayParams {
+  safe: string;
+  payment: string;
+}
+
+export interface GnosisSafeDeploymentStatusParams {
+  owner: Address;
+  safe: Address;
+}
+
+export interface GnosisSafeStatusPayload
+  extends GnosisSafeDeploymentStatusParams {
+  status: GnosisSafeState;
+  txHash?: string;
 }
 
+// TODO remove when onBlock event works
+let intervalId = null;
+
 export class Gnosis {
-  private readonly provider: Provider;
-  private readonly augur: Augur;
-  private readonly gnosisRelay: IGnosisRelayAPI;
-
-  constructor(provider: Provider, gnosisRelay: IGnosisRelayAPI, augur: Augur) {
-    this.provider = provider;
-    this.augur = augur;
-    this.gnosisRelay = gnosisRelay;
-    this.provider.storeAbiData(abi.GnosisSafe as Abi, "GnosisSafe");
-    this.provider.storeAbiData(abi.ProxyFactory as Abi, "ProxyFactory");
-    this.provider.storeAbiData(abi.GnosisSafeRegistry as Abi, "GnosisSafeRegistry");
+  constructor(
+    private readonly provider: Provider,
+    private readonly gnosisRelay: IGnosisRelayAPI,
+    private readonly augur: Augur,
+    private readonly dependencies: ContractDependenciesGnosis
+  ) {
+
+    // TODO this currently doesn't work - using setInterval as workaround
+    // Check safe status on new block. Possible to wait for a transfer event to show up in the DB if this is problematic.
+    augur
+      .getAugurEventEmitter()
+      .on(SubscriptionEventName.NewBlock, this.onNewBlock);
+
+    this.provider.storeAbiData(abi.GnosisSafe as Abi, 'GnosisSafe');
+    this.provider.storeAbiData(abi.ProxyFactory as Abi, 'ProxyFactory');
+    this.provider.storeAbiData(
+      abi.GnosisSafeRegistry as Abi,
+      'GnosisSafeRegistry'
+    );
+  }
+
+  private safesToCheck: GnosisSafeStatusPayload[] = [];
+
+  // Check for safe creations on new block
+  // At this point we will emit status per safe wallet per block. If this is
+  // too noisy for consumers of the event we can be more clever about it.
+  private onNewBlock = async () => {
+    for (const s of this.safesToCheck) {
+      const status = await this.getGnosisSafeDeploymentStatusViaRelay(s);
+      this.augur.setGnosisStatus(status.status);
+      if ([GnosisSafeState.AVAILABLE, GnosisSafeState.CREATED].includes(status.status)) {
+        const signerAddress = await this.dependencies.signer.getAddress();
+        if (signerAddress === s.owner) {
+          this.augur.setGnosisSafeAddress(ethUtil.toChecksumAddress(s.safe));
+          this.augur.setUseGnosisSafe(true);
+        }
+      }
+
+      // Can only register Contract if the current signer is the safe owner.
+      // Be sure the safe creation transaction has been mined.
+      if (status.status === GnosisSafeState.CREATED) {
```

Comment:
Whats the status of using the actual `to`and `data` args instead of a separate delegatecall tx?

Reply:
@justinbarry? 
Are we going to circle back on this? Can you create an issue to track the work needed here?

Reply:
@nuevoalex We are setting up the safe on create now. No need for the delegate call but the status will skip from WAITING_FOR_FUNDS to AVAILABLE unless the separate delegate call.